src/js/lib/project-manager/src/build.ts

```diff
@@ -0,0 +1,107 @@
+const fss = require('fs')
+const fs = fss.promises
+
+const tar = require('tar')
+const os = require('os')
+const path = require('path')
+const unzipper = require('unzipper')
+const url = require('url')
+const { http } = require('follow-redirects')
+
+const thisPath = path.resolve(__dirname)
+const root = path.resolve(thisPath, '..', '..', '..', '..', '..')
+const distPath = path.resolve(root, 'dist', 'bin')
+const buildInfoPath = path.resolve(root, 'dist', 'build.json')
+const { get_project_manager_path } = require('./../../../../../build/paths')
+
+async function get_build_config() {
+    let exists = fss.existsSync(buildInfoPath)
+    if (exists) {
+        let configFile = await fs.readFile(buildInfoPath)
+        return JSON.parse(configFile)
+    }
+}
+
+async function get_target_url(): Promise<string> {
+    const config = await get_build_config()
+    const target_platform = config.target
+    console.log('webpack target ' + target_platform)
+    const version = '0.1.2-rc.18'
+    const base_url = `https://github.com/enso-org/enso-staging/releases/download/enso-${version}/enso-project-manager-${version}`
+    switch (target_platform) {
+        case 'linux':
+            return `${base_url}-linux-amd64.tar.gz`
+        case 'macos':
+            return `${base_url}-macos-amd64.tar.gz`
+        case 'win':
+            return `${base_url}-windows-amd64.zip`
+        default:
+            throw 'UnsupportedPlatform: ' + target_platform
+    }
+}
+
+function project_manager_path() {
+    return get_project_manager_path(distPath)
+}
+
+function decompress_project_manager(source_file_path, target_folder) {
+    let decompressor
+    if (source_file_path.toString().endsWith('.zip')) {
+        decompressor = unzipper.Extract({ path: target_folder })
+    } else {
+        decompressor = tar.x({
+            strip: 1,
+            C: target_folder,
+        })
+    }
+    fss.createReadStream(source_file_path)
+        .pipe(decompressor)
+        .on('finish', () => {
+            const bin_path = project_manager_path()
+            if (os.platform() != 'win32') {
```

Comment:
we are using `win32` name here, and in other places we were renaming osses to our names. Maybe lets create a single place with OS handling and ourrn naes, preferably encoded as enums (enums in JS are just objects with fields being strings, like `os = {windows:'windows', macos:'macos', ...}`

Reply:
This would be touching many places of our build system. I think we should do an extra refactor step and think about what we want to have as our command line input and then normalise our internal handling based on that.